apps/smbmount.sh

```diff
@@ -0,0 +1,288 @@
+#!/bin/bash
+
+# T&M Hansson IT AB © - 2020, https://www.hanssonit.se/
+
+# shellcheck disable=2034,2059
+true
+# shellcheck source=lib.sh
+. <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/master/lib.sh)
+
+# Check for errors + debug code and abort if something isn't right
+# 1 = ON
+# 0 = OFF
+DEBUG=0
+debug_mode
+
+# Check if root
+root_check
+
+# install cifs-utils
+install_if_not cifs-utils
+
+# secure fstab
+if [ "$(stat -c %a /etc/fstab)" != "600" ]
+then
+    chmod 600 /etc/fstab
+fi
+
+# functions
+add_mount() {
+# check if mounting slots are available
+if [ -n "$(grep /mnt/smbshares/1 /etc/fstab)" ] && [ -n "$(grep /mnt/smbshares/2 /etc/fstab)" ] && [ -n "$(grep /mnt/smbshares/3 /etc/fstab)" ]
+then
+    msg_box "No mounting slots available. Please delete one SMB-Mount."
+    return
+fi
+# enter smb-server and share name
+while true
+do
+    SERVER_SHARE_NAME=$(whiptail --inputbox "Please Enter the Server and Share-Name like this:\n//Server/Share" "$WT_HEIGHT" "$WT_WIDTH" 3>&1 1>&2 2>&3)
+    if [[ "no" == $(ask_yes_or_no "Is this correct? $SERVER_SHARE_NAME") ]]
+    then
+        msg_box "It seems like your weren't satisfied by the PATH you entered. Please try again."
+    else
+        SERVER_SHARE_NAME=${SERVER_SHARE_NAME// /\\040}
+        break
+    fi
+done
+# enter smb-user
+while true
+do
+    SMB_USER=$(whiptail --inputbox "Please enter the username of the SMB-user" "$WT_HEIGHT" "$WT_WIDTH" 3>&1 1>&2 2>&3)
+    if [[ "no" == $(ask_yes_or_no "Is this correct? $SMB_USER") ]]
+    then
+        msg_box "It seems like your weren't satisfied by the SMB-user you entered. Please try again."
+    else
+        break
+    fi
+done
+#enter the password of the smb-user
+while true
+do
+    SMB_PASSWORD=$(whiptail --inputbox "Please enter the password of the SMB-user $SMB_USER" "$WT_HEIGHT" "$WT_WIDTH" 3>&1 1>&2 2>&3)
+    if [[ "no" == $(ask_yes_or_no "Is this correct? $SMB_PASSWORD") ]]
+    then
+        msg_box "It seems like your weren't satisfied by the password for the SMB-User you entered. Please try again."
+    else
+        break
+    fi
+done
+# write everything to /etc/fstab, mount and connect external storage
+count=1
```

Comment:
Maybe this is the cause why the suggested change doesn't work.
It should be count="1"
And then probably
count=$($count+1) works

Reply:
this is fixed